spyder/plugins/workingdirectory/plugin.py

```diff
@@ -238,3 +242,534 @@ def chdir(self, directory, browsing_history=False,
         except OSError:
             self.history.pop(self.histindex)
         self.refresh_plugin()
+
+class BreadcrumbsAddressBar(QtWidgets.QFrame):

```

Comment:
Please move this class to `spyder/plugins/workingdirectory/widgets/breadcrumb.py` and rename it to `BreadcrumbAddressBar`.

Reply:
And please also follow our convention of importing Qt stuff like this:

```
from qtpy.QtCore import Signal
from qtpy.QtWidgets import QFrame, QHBoxLayout, ...
```

etc